pkg/eval/realloader_test.go

```diff
@@ -0,0 +1,185 @@
+package eval
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func TestFilterResources(t *testing.T) {
+	inputResources := []schema.GroupVersionResource{
+		{
+			Group:    "",
+			Version:  "v1",
+			Resource: "pods",
+		},
+		{
+			Group:    "",
+			Version:  "v1",
+			Resource: "deployments",
+		},
+		{
+			Group:    "",
+			Version:  "v1",
+			Resource: "persistentvolumeclaims",
+		},
+		{
+			Group:    "config.openshift.io",
+			Version:  "v1",
+			Resource: "clusteroperators",
+		},
+	}
+
+	tests := []struct {
+		name              string
+		includeAll        bool
+		includedGKS       []schema.GroupKind
+		excludedGKS       []schema.GroupKind
+		expectedResources []schema.GroupVersionResource
+	}{
+		{
+			name:        "Include all resources",
+			includeAll:  true,
+			includedGKS: nil,
+			excludedGKS: nil,
+			expectedResources: []schema.GroupVersionResource{
+				{
+					Group:    "",
+					Version:  "v1",
+					Resource: "pods",
+				},
+				{
+					Group:    "",
+					Version:  "v1",
+					Resource: "deployments",
+				},
+				{
+					Group:    "",
+					Version:  "v1",
+					Resource: "persistentvolumeclaims",
+				},
+				{
+					Group:    "config.openshift.io",
+					Version:  "v1",
+					Resource: "clusteroperators",
+				},
+			},
+		},
+		{
+			name:              "Include nothing",
+			includeAll:        false,
+			includedGKS:       nil,
+			excludedGKS:       nil,
+			expectedResources: nil,
+		},
+		{
+			name:       "Include only some resources",
+			includeAll: false,
+			includedGKS: []schema.GroupKind{
+				{
+					Group: "",
+					Kind:  "Pod",
+				},
+				{
+					Group: "",
+					Kind:  "Deployment",
+				},
+			},
+			excludedGKS: nil,
+			expectedResources: []schema.GroupVersionResource{
+				{
+					Group:    "",
+					Version:  "v1",
+					Resource: "pods",
+				},
+				{
+					Group:    "",
+					Version:  "v1",
+					Resource: "deployments",
+				},
+			},
+		},
+		{
+			name:        "Exclude some resources",
+			includeAll:  true,
+			includedGKS: nil,
+			excludedGKS: []schema.GroupKind{
+				{
+					Group: "",
+					Kind:  "Pod",
+				},
+				{
+					Group: "",
+					Kind:  "Deployment",
+				},
+			},
+			expectedResources: []schema.GroupVersionResource{
+				{
+					Group:    "",
+					Version:  "v1",
+					Resource: "persistentvolumeclaims",
+				},
+				{
+					Group:    "config.openshift.io",
+					Version:  "v1",
+					Resource: "clusteroperators",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
```

Comment:
I would prefer the version, there each test case is just a plain old code, instead of the loop over expected inputs and outputs. The reason is that in this case, if some test fails, it's hard to in which exactly scenario it is and one needs to spend more time unpacking than necessary. The extra boilerplate can be extracted to some function to DRY the code, so the total lines of code might not necessary grow that much.

Reply:
Nm, I haven't noticed the `t.Run` wrapper: this addresses my original concerns.